### PR TITLE
Move `StartupHooks` tests and remaining uses of PortableApp project to pre-built test assets

### PR DESCRIPTION
**Pull request by elinor-fung**
- Make `HelloWorld` project handle same options as `PortableApp` and switch tests to use it
- Split `Breadcrumbs` tests out of `NativeHostApis`
- Move `StartupHook` and `StartupHookWithAssemblyResolver` to pre-built assets
- Delete `PortableApp` test project

### CODE REVIEW

File: src/installer/tests/Assets/Projects/StartupHookWithAssemblyResolver/StartupHookWithAssemblyResolver.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+  </PropertyGroup>
+
+  <!-- Copy the preloaded base type to the startup hook's output
+       directory to make it easy to find. -->
+  <ItemGroup>
+    <ProjectReference Include="../SharedLibrary/SharedLibrary.csproj" />
```
Comment (am11) [inline src/installer/tests/Assets/Projects/StartupHookWithAssemblyResolver/StartupHookWithAssemblyResolver.csproj:10]: nit: we normally use backslashes for path repo-wide (and MSBuild takes care of transformations).

File: src/installer/tests/Assets/TestProjects/PortableApp/PortableApp.csproj
Comment (am11) [inline src/installer/tests/Assets/TestProjects/PortableApp/PortableApp.csproj:1]: (on the topic of deleting stuff from install tests) Looks like all scripts under `src/installer/tests/scripts/linux-test` are also candidate of deletion (they do not seem to work with modern OS/dotnet versions). :)

**Review by vitek-karas — APPROVED** [review]
Thanks!

File: src/installer/tests/Assets/Projects/StartupHookWithAssemblyResolver/StartupHookWithAssemblyResolver.csproj (hunk 1 above)
Comment (elinor-fung) [reply src/installer/tests/Assets/Projects/StartupHookWithAssemblyResolver/StartupHookWithAssemblyResolver.csproj:10]:
```suggestion
    <ProjectReference Include="..\SharedLibrary\SharedLibrary.csproj" />
```

**Pull request merged**